### PR TITLE
convert profile components

### PR DESCRIPTION
**Pull request by yulia-dnistrian**
Description:

Linting:

- [ ] No linting errors

Tests:

- [ ] E2E tests (npm test run with `e2e`)
- [ ] Manual tests

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

File: components/src/ProfileEmailInfo/profileemailinfo.main.tsx [hunk 1]
```diff
@@ -72,46 +80,38 @@ class ProfileemailinfoMain extends React.Component<ProfileemailinfoMainProps, Pr
...
-        console.error(error.message);
-      });
+        },
+      },
+    );
+
+    root.defaultprofile.emails.emailform({
+      emailId: '',
+      email,
+    }).fetch({}).then(() => {
```
Comment (dusanradovanovic) [inline components/src/ProfileEmailInfo/profileemailinfo.main.tsx]: Can you change promise chain (than/catch) to await with regular try/catch.

File: components/src/ProfileInfo/profileInfo.main.tsx [hunk 2]
```diff
@@ -86,58 +94,22 @@ class ProfileInfoMain extends React.Component<ProfileInfoMainProps, ProfileInfoM
...
-            // eslint-disable-next-line no-console
-              console.error(error.message);
-            });
-        }).catch((error) => {
-        // eslint-disable-next-line no-console
-          console.error(error.message);
-        });
+    const { onChange } = this.props;
+    const root = await this.client.root().fetch({ defaultprofile: {} });
+    this.client.profile(root.defaultprofile.uri).update({
```
Comment (dusanradovanovic) [inline components/src/ProfileInfo/profileInfo.main.tsx]: You are missing `await` here.

File: components/src/ProfileInfo/profileInfo.main.tsx [hunk 3]
```diff
@@ -86,58 +94,22 @@ class ProfileInfoMain extends React.Component<ProfileInfoMainProps, ProfileInfoM
...
+      givenName: firstName,
+      familyName: lastName,
+      profileId: '',
     });
+    onChange();
+    this.cancel();
+    onChange();
```
Comment (dusanradovanovic) [inline components/src/ProfileInfo/profileInfo.main.tsx]: You are calling onChange() twice. Is this intentional or a mistake?

Comment (yulia-dnistrian) [reply components/src/ProfileInfo/profileInfo.main.tsx]: There was some issue with updating givenName and familyName. Sometimes after calling onChange() those fields are not updated. That is why I call onChange() twice. In this case, everything works correctly.

**Pull request merged**